src/compiler/transformers/ts.ts

```diff
@@ -918,11 +930,12 @@ namespace ts {
             // If the class does not contain nodes that require a synthesized constructor,
             // accept the current constructor if it exists.
             if (!hasInstancePropertyWithInitializer && !hasParameterPropertyAssignments) {
+                getInitializedProperties(node, /*isStatic*/ false, hoistedComputedNames, decoratedMembers);

```

Comment:
The fact we don't use the return value here indicates `getInitializedProperties` now has too many responsibilities.

Reply:
If I rename it to `collectPropertyTransformationResults` and make it output all three things through output array arguments, does it become acceptable? Because it certainly doesn't _only_ apply for initialized properties anymore, and all the work still needs to be done.

Reply:
Possibly, if you are able to pass the results around rather than recalculate them. In general though I'd rather see smaller and more focused methods, e.g.:

- `getInitializedProperties(node, isStatic)`
- `getHoistedComputedNames(properties)`
- `getDecoratedMembers(properties)`